pkg/cloudprovider/aws/block_store.go

```diff
@@ -43,6 +49,32 @@ type blockStore struct {
 }
 
 func getSession(config *aws.Config) (*session.Session, error) {
+	// we need to override the entire credential provider chain in order to set
+	// higher timeout for ec2metata service as this can cause problems with kube2iam
+	metaSess, err := session.NewSession()
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	creds := credentials.NewChainCredentials(
+		[]credentials.Provider{
+			// this is the order in which credential providers will be tried
+			&credentials.EnvProvider{},
+			&credentials.SharedCredentialsProvider{},
+			&ec2rolecreds.EC2RoleProvider{
+				Client: ec2metadata.New(
+					metaSess,
+					&aws.Config{
+						HTTPClient: &http.Client{
+							Timeout: 30 * time.Second,
+						},
+					},
+				),
+			},
+		},
+	)
+	config.WithCredentials(creds)
```

Comment:
We can't do this as-is because it overrides the default credentials loading order that we get from https://github.com/aws/aws-sdk-go/blob/fb9d53b0db7e801eb0d4fa021f5860794d845da3/aws/session/session.go#L416-L469. But I do think we can come up with a solution. We can either copy the aws session logic and modify it as needed, or (if we're ok changing the timeout for all ec2 calls), I think you could just set `config.HTTPClient` and it should trickle down to the ec2 role provider's metadata client.

Reply:
If we're ok with a 30 second timeout across the board then adding `config.WithHTTPClient` to `getSession` seems to work:

```golang
func getSession(config *aws.Config) (*session.Session, error) {
        config.WithHTTPClient(&http.Client{Timeout: 30 * time.Second})
	sess, err := session.NewSession(config)
	if err != nil {
		return nil, errors.WithStack(err)
	}

	if _, err := sess.Config.Credentials.Get(); err != nil {
		return nil, errors.WithStack(err)
	}

	return sess, nil
}
```